statistics/handle/handle.go

```diff
@@ -2043,44 +2024,12 @@ func (h *Handle) getGlobalStatsReader(snapshot uint64) (reader *statsReader, err
 			h.mu.Unlock()
 		}
 	}()
-	return h.getStatsReader(snapshot, h.mu.ctx.(sqlexec.RestrictedSQLExecutor))
+	return statistics.GetStatsReader(snapshot, h.mu.ctx.(sqlexec.RestrictedSQLExecutor))
 }
 
-func (h *Handle) releaseGlobalStatsReader(reader *statsReader) error {
+func (h *Handle) releaseGlobalStatsReader(reader *statistics.StatsReader) error {
 	defer h.mu.Unlock()
-	return h.releaseStatsReader(reader, h.mu.ctx.(sqlexec.RestrictedSQLExecutor))
-}
-
-func (h *Handle) getStatsReader(snapshot uint64, exec sqlexec.RestrictedSQLExecutor) (reader *statsReader, err error) {
-	failpoint.Inject("mockGetStatsReaderFail", func(val failpoint.Value) {
-		if val.(bool) {
-			failpoint.Return(nil, errors.New("gofail genStatsReader error"))
-		}
-	})
-	if snapshot > 0 {
-		return &statsReader{ctx: exec, snapshot: snapshot}, nil
-	}
-	defer func() {
-		if r := recover(); r != nil {
-			err = fmt.Errorf("getStatsReader panic %v", r)
-		}
-	}()
-	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnStats)
-	failpoint.Inject("mockGetStatsReaderPanic", nil)
-	_, err = exec.(sqlexec.SQLExecutor).ExecuteInternal(ctx, "begin")
-	if err != nil {
-		return nil, err
-	}
-	return &statsReader{ctx: exec}, nil
-}
-
-func (h *Handle) releaseStatsReader(reader *statsReader, exec sqlexec.RestrictedSQLExecutor) error {
```

Comment:
The function asks that `reader.ctx` and `exec` must be the same one, which is error-prone. So I changed the function to `func ReleaseStatsReader(reader *StatsReader) error`.

Reply:
After discussing with @qw4990, it is better to rename `releaseStatsReader` to `(*StatsReader).Close`, which is typical golang style.